File: app/src/main/java/co/ab180/awesomeapp/ui/launcher/LauncherActivity.java
package co.ab180.awesomeapp.ui.launcher;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import co.ab180.awesomeapp.ui.main.MainActivity;
import kotlinx.coroutines.GlobalScope;

public class LauncherActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Handler handler = new Handler();
        handler.postDelayed(new Runnable() {
            @Override
            public void run() {
                startActivity(new Intent(getBaseContext(), MainActivity.class));
            }
        }, 1000);
    }
}

File: app/src/main/java/co/ab180/awesomeapp/ui/launcher/LauncherActivity.kt
package co.ab180.awesomeapp.ui.launcher

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import co.ab180.awesomeapp.R
import co.ab180.awesomeapp.ui.main.MainActivity
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch

class LauncherActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_launcher)

        GlobalScope.launch(Dispatchers.Main) {
            delay(1000)
            startActivity(Intent(baseContext, MainActivity::class.java))
            finish()
        }
    }
}